date_range/models/date_range.py

```diff
@@ -52,6 +58,45 @@ def _check_company_id_type_id(self):
                     _('The Company in the Date Range and in '
                       'Date Range Type must be the same.'))
 
+    @api.constrains('parent_id', 'date_start', 'date_end')
+    def _validate_child_range(self):
+        for this in self:
+            if not this.parent_id:
+                continue
+            start = this.parent_id.date_start <= this.date_start
```

Comment:
You can make your code much more readable by renaming 'start' to date_start_valid end 'end' to date_end_valid. Then, if date_start_valid and date_end_valid, just continue and you save an identation level. And you do not then need the strangely named child_range variable.